src/Common/HostResolvePool.h

```diff
@@ -188,7 +219,7 @@ class HostResolver : public std::enable_shared_from_this<HostResolver>
 
     std::mutex mutex;
 
-    Poco::Timestamp last_resolve_time TSA_GUARDED_BY(mutex);
+    Poco::Timestamp last_resolve_time TSA_GUARDED_BY(mutex) = Poco::Timestamp::TIMEVAL_MIN;
```

Comment:
just to be sure that HostResolver::update is called in c-tor even if history is `0`.